sympy/combinatorics/polyhedron.py

```diff
@@ -533,6 +534,7 @@ def rotate(self, perm):
 
         >>> from sympy.combinatorics import Polyhedron, Permutation
         >>> from sympy.combinatorics.polyhedron import cube
+        >>> cube = cube.copy()
```

Comment:
If these objects are supposed to be imported and used like this it seems they shouldn't be mutable. Actually it looks like they are Basic instances, which shouldn't ever be mutable. I think either it should not be Basic, or the methods like `rotate` should return new instances.

Reply:
I thought of making them immutable but that would be a backward compatibility break, and I wasn't sure if it was worth it

Reply:
We can open a new issue about it. For now, this PR seems to fix the failures from the combinatorics and group modules, so I'm +1 to merge it.